lib/shared/addon/components/cluster-driver/driver-oracleoke/component.js

```diff
@@ -0,0 +1,448 @@
+    const mode = get(this, 'vcnCreationMode');
+
+    if (mode === 'Quick') {
+      return false;
+    } else if (mode === 'Existing') {
+      // Driver will use the same compartment as the cluster if not set.
+      return (get(this, 'config.vcnName') && get(this, 'config.loadBalancerSubnetName1')) ? false : true;
+    } else if (mode === 'Custom') {
+      return (get(this, 'config.subnetAccess') && get(this, 'config.vcnCidr')) ? false : true;
+    }
+
+    return true;
+  }),
+  canCreateCluster: computed('config.nodeShape', function() {
+    return get(this, 'config.nodeShape') ? false : true;
+  }),
+
+  loadLanguage(lang) {
+    const translation = languages[lang] || languages['en-us'];
+    const intl = get(this, 'intl');
+
+    intl.addTranslations(lang, translation);
+    intl.translationsFor(lang);
+    set(this, 'refresh', false);
+  },
+  // Add custom validation beyond what can be done from the config API schema
+  validate() {
+    // Get generic API validation errors
+    this._super();
+    var errors = get(this, 'errors') || [];
+
+    if (!get(this, 'cluster.name')) {
+      errors.push('Name is required');
+    }
+
+    const tenancyId = get(this, 'config.tenancyId');
+
+    if (!tenancyId.startsWith('ocid1.tenancy')) {
+      errors.push('A valid tenancy OCID is required');
+    }
+
+    const compartmentId = get(this, 'config.compartmentId');
+
+    if (!compartmentId.startsWith('ocid1.compartment')) {
+      errors.push('A valid compartment OCID is required');
+    }
+
+    const userOcid = get(this, 'config.userOcid');
+
+    if (!userOcid.startsWith('ocid1.user')) {
+      errors.push('A valid user OCID is required');
+    }
+
+    // TODO Add more specific errors
+
+    // Set the array of errors for display,
+    // and return true if saving should continue.
+    if (get(errors, 'length')) {
+      set(this, 'errors', errors);
+
+      return false;
+    } else {
+      set(this, 'errors', null);
+
+      return true;
+    }
+  },
+  willSave() {
+    if (get(this, 'mode') === 'new') {
+      if (get(this, 'config.nodeImage') === '') {
+        set(this, 'config.nodeImage', imageMap['Oracle-Linux-7.6']);
+      }
+      if (get(this, 'config.vcnCompartmentId') === '') {
+        set(this, 'config.vcnCompartmentId', get(this, 'config.compartmentId'));
+      }
+
+
+      if (get(this, 'config.subnetAccess') === 'public') {
+        set(this, 'config.enablePrivateNodes', false);
+      } else {
+        set(this, 'config.enablePrivateNodes', true);
+      }
+    }
+
+    return this._super(...arguments);
+  },
+  k8sUpgradableTo(currVer, toVer) {
+    if (typeof currVer !== 'string') {
+      return false;
+    }
+    if (typeof toVer !== 'string') {
+      return false;
+    }
+
+    currVer = currVer.substr(1);
+    toVer = toVer.substr(1);
+    currVer = currVer.split('.');
+    toVer = toVer.split('.');
+
+    var len = Math.max(currVer.length, toVer.length);
+
+    for (var i = 0; i < len; i++) {
```

Comment:
It doesn't necessarily matter for any current versions but these will not work right because the parts are still strings and e.g. `"2" > "10"`. 

Also see https://github.com/rancher/ui/blob/master/lib/shared/addon/utils/parse-version.js#L102